### PR TITLE
Allow all binding methods on RX

### PR DESCRIPTION
**Pull request by CapnBry**
Removes the distinction between flashing with a bind phrase and traditional binding. This allows users to flash with a phrase, press the receiver's button, 3-plug, use the Betaflight "bind rx" button, or use the webui to change the binding, regardless of how they flashed.

### New Stuff

* Receivers with buttons now have a "bind" button (hold for 1.5s). Wifi start moved to 5s, reset config moved to 12s.
* Lua / Webui option to forget binding information on every boot. "Bind Storage: Volatile"
* Internal: TX module now supports CRSF bind command from the handset
* Internal: RX now supports CRSF bind command from OTA

### Receiver Requirements

TLDR: Any method can be used to change the bind phrase, and the last method used always takes over.

1. A receiver with no flashed bind phrase or current binding info will boot to binding mode
2. On the first boot after flashing with a binding phrase, the receiver will always replace its bind information with the flashed ID. i.e. flashing with a phrase set always binds to that phrase.
3. A user may 3-plug, use the betaflight Bind RX button, or hold the boot button  _after boot_ for 1.5 seconds to enter binding mode
  * If the receiver is unplugged without receiving a bind, the previous bind will still remain
4. Any action from 3 to enter binding mode deliberately will prevent the receiver from entering wifi mode-- the receiver will stay in binding mode until repowered or bound
  * If the receiver is already in wifi mode, it will leave wifi mode and enter binding mode indefinitely
5. A user may use the webui to enter a new binding phrase. This will be handled the same as 2
6. A user may hold the boot button for 15 seconds to erase all receiver configuration (except Regulatory Domain).
  * Regardless of if flashed with a bind phrase, the receiver will boot to binding mode as in 1
7. A user will be presented with a Lua / webui option to never save binding information (per receiver) and always boot to bind mode.
8. Loan Model will be removed. Loaning is done by just binding using any method in 3 just like normal binding. This "loan" can not be revoked. The user simply rebinds to their transmitter again.
  * The Loan Model button in Lua will be replaced with an option to force the RX into binding mode even though the connection is established

### TX Requirements

The TX operation does not change

1. Flashing with a bind phrase uses that ID
2. Overriding the ID in the webui replaces any flashed ID
  * Flashing with a bind phrase after overriding uses the new flashed ID
3. If no ID has been set, the TX will use a persistent autogenerated unique ID specific to that device

### Webui Changes

Because the binding information for the RX is no longer "Runtime Options" it made sense to move this to model settings. The binding phrase input box now also takes 4-6 comma-separated numbers to allow binding to an ID without knowing the bind phrase.

image

### Reset Config

The 15s "reset configuration" button resets **everything**: bind phrase, home wifi, serial baud, etc. This removes any chance the user may have flashed with some weird setting like 416666 baud. However, STM32 receivers have no defaults to fall back on so they lose their binding phrase, but nothing else.

PROBLEMS:
* What to default regulatory domain to? Currently it carries over the reg domain, as there is no way to default this. 

### Bind Command

Similar to #2523, the TX will now look for and process a bind command addressed to it, and the RX can receive the command OTA. This allows directly calling the Lua's [BIND] button without having to load the entire config looking at it. EdgeTX could add this command to their UI, or a dedicated Bind widget. Sample code

```lua 
local bindSent

local function run(event)
    local CRSF_FRAMETYPE_COMMAND = 0x32
    local CRSF_ADDRESS_RADIO_TRANSMITTER = 0xEA
    local CRSF_ADDRESS_CRSF_RECEIVER = 0xEC
    local CRSF_ADDRESS_CRSF_TRANSMITTER = 0xEE
    local CRSF_COMMAND_SUBCMD_RX = 0x10
    local CRSF_COMMAND_SUBCMD_RX_BIND = 0x01

    if event == EVT_VIRTUAL_ENTER then
        crossfireTelemetryPush(CRSF_FRAMETYPE_COMMAND,
            { CRSF_ADDRESS_CRSF_TRANSMITTER,  -- or CRSF_ADDRESS_CRSF_RECEIVER 
              CRSF_ADDRESS_RADIO_TRANSMITTER,
              CRSF_COMMAND_SUBCMD_RX, CRSF_COMMAND_SUBCMD_RX_BIND }
        )
        bindSent = true
    end

    lcd.clear()
    lcd.drawText(0, 0, bindSent and "Bind sent" or "Press enter to bind")

    return 0
  end

  return { run = run }
```

### Unrelated Changes

* Buttons on any non-unified targets (STM32) were disabled, consulted with @PK to return them
* Targets with external EEPROM over I2C have not been able to load their configuration since #2425
* Config update from v7 was broken, it didn't do anything it just effectively reset to defaults. Fixed.

## Tested

* EP2 RX + TX16S Internal
* EP2 RX + Namimno OLED
* R9MM + ES900TX

Thanks to everyone who contributed to the requirements for this PR: the ExpressLRS development team, ItsBlunty, Joshua Bardwell, Mads Tech, RC Video Reviews, and a big bottle of single malt.

### CODE REVIEW

**Comment by CapnBry** [comment]
Updated with #2523 and supports entering binding mode when BF button is pressed even if receiver is connected.

File: src/lib/BUTTON/devButton.cpp [hunk 1]
```diff
@@ -29,9 +30,10 @@ static constexpr struct {
     bool pressType;
     uint8_t count;
     action_e action;
-} button_actions[2] = {
-    {true, 3, ACTION_START_WIFI},
-    {true, 7, ACTION_RESET_REBOOT}
+} button_actions[] = {
+    {true, 1, ACTION_BIND},
+    {true, 7, ACTION_START_WIFI},
+    {true, 14, ACTION_RESET_REBOOT}
```
Comment (pkendall64) [inline src/lib/BUTTON/devButton.cpp:36]: Don't forget that these are in half second increments + 1 extra half second for zero.
So your times are 1s for binding, 4 seconds for wifi and 7.5 seconds for reboot.

Comment (CapnBry) [reply src/lib/BUTTON/devButton.cpp]: You are 100% correct, no wonder they seemed so close together! I had forgotten.

**Comment by pitts-mo** [comment]
Thanks CapnBry :-)

Buddy Ol' Pal ... 
Think you might find time to squeeze some of this binding goodness over into VRX backpack? 

I would like to prevent an Analog RX in HDZero module bay from loosing bind if you boot to another source three times in a row when using HDZero Goggle's AnalogRX Power = Auto option and this web based control would be helpful to manually set binding phrase/UID without entering at build time. 

I also read this now falls back to the last binding :-) but still wonder if it might still be desirable to make the three plug timeout a binding page option with 0 seconds being disabled and then use 15 second button press to clear it as needed. 

Again, Thank you so much.
-p

**Comment by CapnBry** [comment]
> Think you might find time to squeeze some of this binding goodness over into VRX backpack?

If you have problems with how binding works on the VRX Backpack, that sounds like something you should bring up on the backpack github instead of starting an off topic discussion here ;)

File: src/user_defines.txt [hunk 2]
```diff
@@ -44,6 +44,7 @@
 ### COMPATIBILITY OPTIONS: ###
 
 # Use a custom baud rate on the receiver for a KISS v1 FC (which runs at 400000) or any other oddball baud
+# Changing this will prevent flashing via_BetaflightPassthrough, which expects a 400000 baud RX
```
[the remark this answers is not in the document]
Comment (CapnBry) [reply src/user_defines.txt]: I add one line of docs and typo it 😆

File: src/lib/OPTIONS/options.cpp [hunk 3]
```diff
@@ -24,6 +24,10 @@ const char *wifi_ap_ssid = "ExpressLRS RX";
 const char *wifi_ap_password = "expresslrs";
 const char *wifi_ap_address = "10.0.0.1";
 
+// Discriminator value used to determine if the device has been reflashed and therefore
+// the SPIFSS settings are obsolete and the flashed settings should be used in preference
+uint32_t flash_discriminator;
```
[the remark this answers is not in the document]
Comment (CapnBry) [reply src/lib/OPTIONS/options.cpp]: haha To be honest, I completely forgot the binary_configurator supported STM32! In my mind it was build-yourself-only. Updated to move the flash_discriminator into firmwareOptions on all platforms, and patch the discriminator on STM32 (both TX/RX but only compared on the RX).

**Review by pkendall64 — APPROVED** [review]
Awesome!

**Review by deadbytefpv — APPROVED** [review]
Top notch work! Works as advertised!

**Comment by sokil** [comment]
https://github.com/ExpressLRS/ExpressLRS/issues/2864

After removing bind phrase update from saveOptions method updaring of home wifi setting remove binding phrase.

**Pull request merged**